File: ch3/yojkim/ch3/dollar.go
package ch3

type Dollar struct {
	Amount int
}

func NewDollar(amount int) Dollar {
	dollar := Dollar{}
	dollar.Amount = amount

	return dollar
}

func (d Dollar) Times(multiplier int) *Dollar {
	return &Dollar{d.Amount * multiplier}
}

func (d Dollar) Equal(input Dollar) bool {
	return d.Amount == input.Amount
}

File: ch3/gunddol/src/ch3/dollar.java
package ch3;

public class dollar {
    int amount;

    public dollar(int amount){
        this.amount = amount;
    }

    public dollar times(int multiplier) {
        return new dollar(amount * multiplier);
    }

    public boolean equals(Object object){
        dollar dollar = (dollar) object;
        return amount == dollar.amount;
    }
}
